Remove Lifetime specification from Output

ktf opened the pull request:
Remove Lifetime specification from Output

The .lifetime field in Output is to be removed. This includes
cases in which the value is not the default.

---
Stack created with [Sapling](https://sapling-scm.com). Best reviewed with [ReviewStack](https://reviewstack.dev/AliceO2Group/AliceO2/pull/12400).
* #12401
* __->__ #12400
* #12399

ktf committed:
Remove Lifetime specification from Output

The .lifetime field in Output is to be removed. This includes
cases in which the value is not the default.

alibuild commented:
Error while checking build/O2/fullCI for 5654c48597898d436a8504946c6b497832594a48 at 2023-12-05 12:04:
```
## sw/BUILD/O2-latest/log
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
/sw/SOURCES/O2/12400-slc8_x86-64/0/Utilities/DataSampling/src/Dispatcher.cxx:117:37: error: no matching function for call to 'o2::framework::Output::Output(<brace-enclosed initializer list>)'
ninja: build stopped: subcommand failed.
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/12400/5654c48597898d436a8504946c6b497832594a48/build_O2_fullCI/pretty.html).

ktf committed:
Remove default Lifetime::Timeframe when instanciating an Output

This lifetime field will be removed soon, and this is unambiguous, because
it's in any case the default.
